cpp/src/lance/arrow/dataset.cc

```diff
@@ -382,4 +391,52 @@ ::arrow::Result<::arrow::dataset::FragmentIterator> LanceDataset::GetFragmentsIm
   return ::arrow::MakeVectorIterator(fragments);
 }
 
+::arrow::Result<std::shared_ptr<LanceDataset>> LanceDataset::AddColumns(
+    const std::shared_ptr<::arrow::Table>& other,
+    const std::string& on,
```

Comment:
not sure we'll always have the luxury of the same name on both. Can change this later tho i guess

Reply:
we can change the signature to  `AddColumns(other, left_on, right_on)`? I tried, but could not come up with a good function signature. 
 
 Suggestions?

Reply:
`left_on` and `right_on` seems fine. In python we can have optional `on=` if the column name is indeed the same ?

Reply:
Done